train.py

```diff
@@ -6,70 +6,17 @@
 """torchgeo model training script."""
 
 import os
-from typing import Any, cast
+from typing import cast
 
 import lightning.pytorch as pl
+from hydra.utils import instantiate
 from lightning.pytorch import LightningDataModule, LightningModule, Trainer
 from lightning.pytorch.callbacks import EarlyStopping, ModelCheckpoint
 from lightning.pytorch.loggers import CSVLogger, TensorBoardLogger
 from omegaconf import DictConfig, OmegaConf
 
-from torchgeo.datamodules import (
-    BigEarthNetDataModule,
-    ChesapeakeCVPRDataModule,
-    COWCCountingDataModule,
-    DeepGlobeLandCoverDataModule,
-    ETCI2021DataModule,
-    EuroSATDataModule,
-    GID15DataModule,
-    InriaAerialImageLabelingDataModule,
-    LandCoverAIDataModule,
-    LoveDADataModule,
-    NAIPChesapeakeDataModule,
-    NASAMarineDebrisDataModule,
-    Potsdam2DDataModule,
-    RESISC45DataModule,
-    SEN12MSDataModule,
-    So2SatDataModule,
-    SpaceNet1DataModule,
-    TropicalCycloneDataModule,
-    UCMercedDataModule,
-    Vaihingen2DDataModule,
-)
-from torchgeo.trainers import (
-    BYOLTask,
-    ClassificationTask,
-    MultiLabelClassificationTask,
-    ObjectDetectionTask,
-    RegressionTask,
-    SemanticSegmentationTask,
-)
-
-TASK_TO_MODULES_MAPPING: dict[
-    str, tuple[type[LightningModule], type[LightningDataModule]]
-] = {
-    "bigearthnet": (MultiLabelClassificationTask, BigEarthNetDataModule),
-    "byol": (BYOLTask, ChesapeakeCVPRDataModule),
-    "chesapeake_cvpr": (SemanticSegmentationTask, ChesapeakeCVPRDataModule),
-    "cowc_counting": (RegressionTask, COWCCountingDataModule),
-    "cyclone": (RegressionTask, TropicalCycloneDataModule),
-    "deepglobelandcover": (SemanticSegmentationTask, DeepGlobeLandCoverDataModule),
-    "eurosat": (ClassificationTask, EuroSATDataModule),
-    "etci2021": (SemanticSegmentationTask, ETCI2021DataModule),
-    "gid15": (SemanticSegmentationTask, GID15DataModule),
-    "inria": (SemanticSegmentationTask, InriaAerialImageLabelingDataModule),
-    "landcoverai": (SemanticSegmentationTask, LandCoverAIDataModule),
-    "loveda": (SemanticSegmentationTask, LoveDADataModule),
-    "naipchesapeake": (SemanticSegmentationTask, NAIPChesapeakeDataModule),
-    "nasa_marine_debris": (ObjectDetectionTask, NASAMarineDebrisDataModule),
-    "potsdam2d": (SemanticSegmentationTask, Potsdam2DDataModule),
-    "resisc45": (ClassificationTask, RESISC45DataModule),
-    "sen12ms": (SemanticSegmentationTask, SEN12MSDataModule),
-    "so2sat": (ClassificationTask, So2SatDataModule),
-    "spacenet1": (SemanticSegmentationTask, SpaceNet1DataModule),
-    "ucmerced": (ClassificationTask, UCMercedDataModule),
-    "vaihingen2d": (SemanticSegmentationTask, Vaihingen2DDataModule),
-}
+from torchgeo import datamodules, trainers  # noqa: F401
```

Comment:
Why is this needed?

Reply:
Since importing torchgeo doesn't import the datamodules, trainers, etc. Hydra will try to instantiate the module and datamodules objects but they won't be defined. So we need to import datamodules and trainers. Simple tradeoff imo.

Reply:
Actually this is solved because I have to import MisconfigurationException from torchgeo.datamodules